airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -118,10 +131,18 @@ def __init__(
         self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
         self.s3_bucket_name = s3_bucket_name
         self.s3_key_prefix = s3_key_prefix
-        self.aws_conn_id = aws_conn_id
+        if aws_conn_id is not NOTSET:
+            warnings.warn(_DEPRECATION_MSG, DeprecationWarning, stacklevel=3)
+            self.source_aws_conn_id = aws_conn_id
+        else:
+            self.source_aws_conn_id = source_aws_conn_id
+        if dest_aws_conn_id is NOTSET:
+            self.dest_aws_conn_id = self.source_aws_conn_id
+        else:
+            self.dest_aws_conn_id = dest_aws_conn_id
```

Comment:
You'd have to test it, but this may be easier as 

```suggestion
    self.dest_aws_conn_id = dest_aws_conn_id or self.source_aws_conn_id
```

I'm pretty sure (but not positive) that NOTSET returns falsy

Reply:
Because this neither implement `__bool__` or `__len__` it return True.
And if `dest_aws_conn_id` is `None` than it replaced by `self.source_aws_conn_id`, however

https://github.com/apache/airflow/blob/17e8bb7f9e320c97fd737f8786a6b16515f4810e/airflow/providers/amazon/aws/hooks/base_aws.py#L379-L388

Reply:
My mistake.  I wonder if we should some day add a __bool__ to it so we can use it that way.  But that's not really relevant now and I suspect there would be a bit of discussion if NOTSET should be truthy or falsy.  `if NOTSET` should resolve to true, I guess, but having it falsy makes those assignments like the one I suggested really nice.

Anyway.. My mistake, feel free to resolve this one.

Reply:
`NOTSET` it is just a sentinel, for case when we can't use `None`. And since community providers are have min Airflow 2.3 support we could safety use it in all places, rather than implement it per provider.